File: Problem Set 1 - C/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float change;
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);

    // how many coins?
    int coins = round(change * 100);

    // how many quarters?
    int quarters = coins / 25;

    // changing the coins variable to become new. Basically trying to find the change.
    coins = coins - (quarters * 25);

    // how many dimes?
    int dimes = coins / 10;

    // find the change again
    coins = coins - (dimes * 10);

    // how many nickels?
    int nickels = coins / 5;

    // find the change one last time
    coins = coins - (nickels * 5);

    // how many pennies? This would be the leftover amount.
    int pennies = coins;

    printf("%i\n", quarters + dimes + nickels + pennies);
}

File: Problem Set 6 - Python/cash.py
from cs50 import get_float
import math

def change():
    while True:
        change = get_float("Change owed: ")
        if change > 0:
            break
    return change

# Run program and round
# Unlike C, where I declared int, I have to round down using math module
change = change()
coins = change * 100

# How many quarters?
quarters = math.floor(coins / 25)
coins = coins - (quarters * 25)

# How many dimes?
dimes = math.floor(coins / 10)
coins = coins - (dimes * 10)

# How many nickels?
nickels = math.floor(coins / 5)
coins = coins - (nickels * 5)

# What's leftover?
pennies = math.floor(coins)

print(quarters + dimes + nickels + pennies)